lib/ruby_indexer/lib/ruby_indexer/enhancement.rb

```diff
@@ -33,7 +33,7 @@ def clear
 
     sig { params(listener: DeclarationListener).void }
     def initialize(listener)
-      @listener = listener
+      @listener = T.let(listener, RubyIndexer::DeclarationListener)
```

Comment:
How will this eliminate the need for the shim? Tapioca doesn't carry instance variable declarations to the RBIs AFAIK.

Also, Sorbet already knows the right type for `@listener` thanks to the method signature.

Reply:
We can try, but I'm not sure if Tapioca picks up instance variable types.